web/src/engine/websites/Sukima.ts

```diff
@@ -0,0 +1,191 @@
+import { Tags } from '../Tags';
+import icon from './Sukima.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import { FetchJSON, FetchRequest, FetchWindowScript } from '../FetchProvider';
+import type { Priority } from '../taskpool/TaskPool';
+import type { Numeric, Text } from '../SettingsManager';
+import { Key as GlobalKey } from '../SettingsGlobal';
+import * as Common from './decorators/Common';
+
+const pageScript = `
+    new Promise((resolve, reject) => {
+
+            PAGES_INFO.sort((a, b) => {
+                return a.page_number - b.page_number;
+            });
+
+            resolve(PAGES_INFO.map(page => {
+                return {
+                page_number : page.page_number, 
+                page_url : page.page_url,
+                shuffle_map : JSON.parse(page.shuffle_map),
+                blocklen : BLOCKLEN
+                }
+            }));
+     });
+
+`;
+
+type APIManga = {
+    title: string,
+    contents: {
+        stories: APIChapter[]
+    }[]
+}
+
+type APIChapter = {
+    title_code: string,
+    volume: number,
+    story: number,
+    info: {
+        text: string
+    }
+}
+
+type PAGE_INFO = {
+    page_number: number,
+    page_url: string,
+    shuffle_map: SHUFFLE_MAP,
+    blocklen: number
+}
+
+type APIMangas = {
+    items: {
+        title_code: string,
+        title_name: string
+    }[]
+    max_page: number
+}
+
+type APICategories = {
+    rows: {
+        more_btn: {
+            link: string
+        }
+    }[]
+}
+
+type SHUFFLE_MAP = Array<Array<number>>;
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('sukima', `Sukima`, 'https://www.sukima.me', Tags.Language.Japanese, Tags.Media.Manga, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/www\.sukima\.me\/book\/title\/\S+\/$/.test(url);
```

Comment:
Probably better to use `[^/]+` instead of `\S+`, to avoid matches like
[https://www.sukima.me/book/title/x/x/x/x/x/x/](https://www.sukima.me/book/title/x/x/x/x/x/x/)